Assignment 2/test/Vault.t.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "src/Vault.sol";
+import "src/Coll.sol";
+import "yield-utils-v2/token/IERC20.sol";
+
+/// @title Test state with just an initialized `Vault` contract.
+/// @notice This test state sets up a `Vault` with `Coll` as token.
+/// @dev The setup is straightforward and happens in `setUp`.
+/// @author Thomas den Hollander
+abstract contract ZeroState {
+    Vault vault;
+    Coll token;
+
+    event Deposit(address indexed account, uint256 deposited);
+    event Withdrawal(address indexed account, uint256 withdrawn);
+
+    function setUp() public virtual {
+        token = new Coll();
+        vault = new Vault(token);
+    }
+}
+
+/// @notice Test state where the contract has minted some tokens.
+abstract contract MintedTokenState is ZeroState {
+    /// @notice The amount of tokens that the contract should own at the start
+    ///     of tests.
+    uint256 constant tokensMinted = 1_000_000_000_000_000_000;
+
+    function setUp() public virtual override {
+        super.setUp();
+        token.mint(address(this), tokensMinted);
+    }
+}
+
+/// @notice Test state where the contract has deposited some tokens. The amount
+///     of tokens is equal to `tokensMinted`.
+abstract contract DepositedTokenState is MintedTokenState {
+    function setUp() public override {
+        super.setUp();
+        token.approve(address(vault), tokensMinted);
+        vault.deposit(tokensMinted);
+    }
+}
+
+/// @notice Test some simple properties of the contract in the `ZeroState`.
+contract ZeroStateTest is ZeroState, Test {
```

Comment:
if u make Zerostate inherit from Test then u only need to inherit from ZeroState